### PR TITLE
feat: add upgrade guide

### PR DESCRIPTION
**Pull request by lucasfernog**

### CODE REVIEW

**Comment by lucasfernog** [comment]
Includes some changes not yet available but I wanted to get this out ASAP. Also the guide assumes v2 stable is already out and the install instructions does not take the alpha in consideration.

**Comment by lorenzolewis** [comment]
Oh this is amazing! I'll take a look at it later this week and do a copy pass

**Comment by lorenzolewis** [comment]
Started working on this one. Overall love the detail, just trying to group the information a bit more and make it easy to link to specific migration sections.

Could you take a look at this one I reworked and let me know if it's okay: https://deploy-preview-1340--tauri-docs-starlight.netlify.app/2/guide/upgrade-migrate/from-tauri-1/#migrate-to-cli-plugin

I'll wait for feedback before I de-dup and re-work the sections below that.

@FabianLars @simonhyll any feedback on any improvements we could make to make migration easier?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+title: Migrate from Tauri 1.0
+---
+
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+This guide walks you through migrating your Tauri 1.0 application to Tauri 2.0.
+
+## Automated Migration
+
+The Tauri v2 CLI includes a `migrate` command that automates most of the process and helps you finish the migration:
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx:12]: Can we explain what this command does so someone knows what to expect before running it? I need to get the CLI documenter up and running so we can also link to it.

Comment (lucasfernog) [reply src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx:12]: I think we should link to the command's help description instead. What it actually does will change a lot until v2. So far it migrates Cargo.toml and tauri.conf.json (that's mostly all it'll do, but we also want to guide the user on the plugins migration and some other breaking changes).

**Comment by FabianLars** [comment]
> FabianLars simonhyll any feedback on any improvements we could make to make migration easier?

Not yet, nope. Looks good so far :)

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx (hunk 1 above)
Comment (lorenzolewis) [reply src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx:12]: Cool, I can rephrase that a bit and point to the new CLI docs that @FabianLars started in #1365 (will just be a broken link for now I guess).

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 2]
```diff
@@ -0,0 +1,918 @@
...
+- `@tauri-apps/api/updater` module removed. Use `@tauri-apps/plugin-updater` instead [Migration](#migrate-to-updater-plugin)
+- `@tauri-apps/api/window` module removed. Use `@tauri-apps/plugin-window` instead [Migration](#migrate-to-window-plugin)
+
+## Detailed Migration Steps
+
+Common scenarios you may encounter when migrating your Tauri 1.0 app to Tauri 2.0.
+
+### Migrate to App Plugin
+
+{/* TODO: Is the Tauri 1.0 Rust equivalent something like tauri::window::Window::hide? */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Is the Tauri 1.0 Rust equivalent something like `tauri::window::Window::hide` (so we can list it here like the rest of the migration sections).

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 3]
```diff
@@ -0,0 +1,918 @@
...
+```js
+import { show, hide } from '@tauri-apps/plugin-app';
+await hide();
+await show();
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Rust example */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 4]
```diff
@@ -0,0 +1,918 @@
...
+import { fetch } from '@tauri-apps/plugin-http';
+const response = await fetch(
+	'https://raw.githubusercontent.com/tauri-apps/tauri/dev/package.json'
+);
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Code example of this one? */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one? This one only initialises the plugin

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 5]
```diff
@@ -0,0 +1,918 @@
...
+
+```js
+import { sendNotification } from '@tauri-apps/plugin-notification';
+sendNotification('Tauri is awesome!');
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Code example on the Rust side */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 6]
```diff
@@ -0,0 +1,918 @@
...
+
+```js
+import { arch } from '@tauri-apps/plugin-os';
+const architecture = await arch();
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Rust example */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 7]
```diff
@@ -0,0 +1,918 @@
...
+```js
+import { exit, relaunch } from '@tauri-apps/plugin-process';
+await exit(0);
+await relaunch();
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Rust example */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one? Should I copy-paste from the shell example?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 8]
```diff
@@ -0,0 +1,918 @@
...
+import { Command, open } from '@tauri-apps/plugin-shell';
+const output = await Command.create('echo', 'message').execute();
+
+await open('https://github.com/tauri-apps/tauri');
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Does this one need to be this complicated or could we simplify it to be similar to the other Rust code blocks? */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Could this example be simplified to cover the most common use case or best example? Or is this a more involved migration that we need to keep this in on?

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx [hunk 9]
```diff
@@ -0,0 +1,918 @@
...
+
+```js
+import { appWindow } from '@tauri-apps/plugin-window';
+await appWindow.setTitle('Tauri');
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+{/* TODO: Rust example */}
```
Comment (lorenzolewis) [inline src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: Any example for this one?

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Sorry for taking a while on this one, it was a long (but very good) one. Put in a few comments just to fill in a few missing pieces and then I think should be good to merge.

Excited on this one because we should be able to easily add to this for different scenarios as we get feedback for any rough spots.

File: src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx (hunk 2 above)
Comment (lucasfernog) [reply src/content/docs/2/guide/upgrade-migrate/from-tauri-1.mdx]: The Rust equivalent wasn't changed.

**Review by lorenzolewis — APPROVED** [review]
Absolutely amazing! Thank you @lucasfernog 

I just tweaked the overall page to be called "Upgrading", but still referring to some of actions (like moving towards a plugin) a "migration". Potato/potato, but upgrading felt like the more appropriate word for the overall guide to me.

**Pull request merged**